### PR TITLE
Add Vitest coverage for shared JavaScript utilities

### PR DESCRIPTION
**Pull request by ramram-mf**
## Description

Adds focused Vitest coverage for the shared redesign carousel and CSRF utilities, including the standalone global CSRF entry point.

## What Changed

- Covers carousel indexing, card cloning, indicator state, debouncing, responsive navigation, loop boundaries, and pointer and keyboard interactions.
- Covers CSRF cookie lookup, token minting and failure handling, native form submission, submitter preservation, and global helper exposure.

Related PRs/issues: [TP1-4150](https://mozilla-hub.atlassian.net/browse/TP1-4150), [GitHub issue #16394](https://github.com/MozillaFoundation/foundation.mozilla.org/issues/16394)

### CODE REVIEW

File: foundation_cms/static/js/utils/csrf.test.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+  it.each([
+    ["an unsuccessful response", vi.fn().mockResolvedValue({ ok: false })],
+    ["a network failure", vi.fn().mockRejectedValue(new Error("offline"))],
+  ])("returns an empty token after %s", async (_label, fetchMock) => {
+    vi.stubGlobal("fetch", fetchMock);
+
+    await expect(ensureCsrfToken()).resolves.toBe("");
+  });
+
```
Comment (mmmavis) [inline foundation_cms/static/js/utils/csrf.test.js:72]: Could we add a test for the case where `/api/csrf/` returns `ok: true` but the `csrftoken` cookie still isn’t set? `ensureCsrfToken()` re-reads the cookie after a successful response, so this is a separate path from the HTTP/network failure cases and should return `""`.

File: foundation_cms/static/js/utils/csrf.test.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    await vi.waitFor(() => expect(submitSpy).toHaveBeenCalledOnce());
+    expect(event.defaultPrevented).toBe(true);
+    expect(form.querySelector("[data-csrf-field]").value).toBe("form-token");
+    const mirror = form.querySelector("[data-csrf-submitter]");
+    expect(mirror.type).toBe("hidden");
+    expect(mirror.name).toBe("action");
+    expect(mirror.value).toBe("share");
+  });
+
```
Comment (mmmavis) [inline foundation_cms/static/js/utils/csrf.test.js:101]: We should also add a test for what happens when `ensureCsrfToken()` fails during form submission? Right now the implementation still calls `form.submit()` even if the token is empty. It would be good to make the intended behavior explicit, since this would likely result in a CSRF failure/403.

File: foundation_cms/static/js/utils/csrf_global.test.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { describe, expect, it } from "vitest";
+import { ensureCsrfToken, getCookie } from "./csrf.js";
+import "./csrf_global.js";
+
+describe("CSRF global entry point", () => {
+  it("exposes the shared helpers on window.FoundationCSRF", () => {
+    expect(window.FoundationCSRF).toEqual({ getCookie, ensureCsrfToken });
```
Comment (mmmavis) [inline foundation_cms/static/js/utils/csrf_global.test.js]:
```suggestion
    expect(window.FoundationCSRF.getCookie).toBe(getCookie);
    expect(window.FoundationCSRF.ensureCsrfToken).toBe(ensureCsrfToken);
```

That way the test still verifies both helpers are exposed without requiring `FoundationCSRF` to have exactly those two properties.

**Review by mmmavis — CHANGES_REQUESTED** [review]
Looks good overall! I left a few small suggestions around edge cases and the global test, but nothing major from me. Thank @ramram-mf  👍🏻

File: foundation_cms/static/js/utils/csrf.test.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+  it.each([
+    ["an unsuccessful response", vi.fn().mockResolvedValue({ ok: false })],
+    ["a network failure", vi.fn().mockRejectedValue(new Error("offline"))],
+  ])("returns an empty token after %s", async (_label, fetchMock) => {
+    vi.stubGlobal("fetch", fetchMock);
+
+    await expect(ensureCsrfToken()).resolves.toBe("");
+  });
```
Comment (mmmavis) [inline foundation_cms/static/js/utils/csrf.test.js:71]: Could we build the mocks inside the test (or pass a factory into `it.each`) so `restoreAllMocks` can’t strip them? Something like:

```js
it.each([
  ["an unsuccessful response", () => vi.fn().mockResolvedValue({ ok: false })],
  ["a network failure", () => vi.fn().mockRejectedValue(new Error("offline"))],
])("returns an empty token after %s", async (_label, createFetchMock) => {
  const fetchMock = createFetchMock();
  vi.stubGlobal("fetch", fetchMock);

  await expect(ensureCsrfToken()).resolves.toBe("");
  expect(fetchMock).toHaveBeenCalledOnce();
});

**Review by mmmavis — CHANGES_REQUESTED** [review]
One last issue. We are so close!

**Review by mmmavis — APPROVED** [review]
🎉
